api/app/Modules/Accounts/Http/Controllers/LibraryController.php

```diff
@@ -27,11 +27,9 @@ public function tracks(Request $request): JsonResponse
         return $this->respondWithPaginator($tracks, new TrackTransformer());
     }
 
-    public function getTrackIds()
+    public function getTrackIds(): JsonResponse
     {
-        $tracks = $this->getUser()->savedTracks()->get(['saveable_id'])->map(function ($result) {
-            return $result->saveable_id;
-        })->unique()->values();
+        $tracks = $this->getUser()->savedTracks()->get(['saveable_id'])->unique()->values();
```

Comment:
Does this change the response?

Reply:
@szainmehdi  This method is not being used so maybe we can remove it ?

Previously it was sending only the saveable_id but now it is sending saveable_id as well as the saveable object

Psalm kept complaining about how saveable_id is not a property on the class, as we technically don't have a saveable model.
But again, I think we can remove this method completely as we are not using it at all, I have checked the frontend code too and no use of this api method `/v1/tracks/ids`

Reply:
We're definitely using it, though it's a little hard to follow the code right now to find where. But, the endpoint definition in the frontend is here
https://github.com/nawhas/nawhas/blob/d04cc2c1b276a1f39a9fdf4cf77ca66581aa36c7/nuxt/api/library.ts#L47-L49

and this method is used here
https://github.com/nawhas/nawhas/blob/d04cc2c1b276a1f39a9fdf4cf77ca66581aa36c7/nuxt/store/library.ts#L34-L53

and that is called in these places
https://github.com/nawhas/nawhas/blob/d04cc2c1b276a1f39a9fdf4cf77ca66581aa36c7/nuxt/middleware/initialize.ts#L11
and https://github.com/search?q=repo%3Anawhas%2Fnawhas%20library%2FgetTrackIds&type=code